tests/system/data_sources/test_teradata.py

```diff
@@ -523,6 +523,30 @@ def test_row_validation_char_pk_to_bigquery():
     id_type_test_assertions(df)
 
 
+@mock.patch(
+    "data_validation.state_manager.StateManager.get_connection_config",
+    new=mock_get_connection_config,
+)
+def test_row_validation_char_comp_field_to_bigquery():
```

Comment:
I wonder if this should be a comparison fields test across all columns rather than restricted to `col_char_2`? That would test for side effects in addition to the "happy path".

I don't think we have any comparison field testing so this could start that process, we could add an issue asking to duplicate this test across other engines?

Reply:
good idea - will add this --comp-fields test to all TD columns